### PR TITLE
Changing "doesExists" to "doesExist"

### PR DESCRIPTION
**Pull request by Borales**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

### CODE REVIEW

**Comment by kamilmysliwiec** [comment]
PR looks good! However, it introduces a breaking change so we'll have to wait till the next major release

**Review by micalevisk — COMMENTED** [review]
I guess you could add this method:

```js
/**
 * @deprecated Use the `doesExist` method instead.
 */
doesExists = this.doesExist;
```

so this PR will be backwards-compatible :smile:

File: lib/scheduler.registry.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export class SchedulerRegistry {
   private readonly timeouts = new Map<string, any>();
   private readonly intervals = new Map<string, any>();
 
-  doesExists(type: 'cron' | 'timeout' | 'interval', name: string) {
+  doesExist(type: 'cron' | 'timeout' | 'interval', name: string) {
```
Comment (micalevisk) [inline lib/scheduler.registry.ts:11]:
```suggestion
  /**
   * @deprecated Use the `doesExist` method instead.
   */
  // TODO(v2): drop this.
  doesExists = this.doesExist;
  doesExist(type: 'cron' | 'timeout' | 'interval', name: string) {
```

**Review by micalevisk — APPROVED** [review]

**Comment by micalevisk** [comment]
actually @Borales can you apply the changes I've made on PR #854 here as well? otherwise devs won't get that `@deprecated` notice

**Comment by Borales** [comment]
@micalevisk done

**Review by micalevisk — APPROVED** [review]

**Comment by kamilmysliwiec** [comment]
lgtm

**Pull request merged**